src/sgl/device/shader.cpp

```
@@ -20,12 +21,72 @@
 
+static bool write_module_bytes_to_cache(
+    const std::string& module_name,
+    const std::filesystem::path& cache_path,
+    const std::vector<uint8_t>& data
+)
+{
+    std::filesystem::path tmp_path = cache_path;
+    std::random_device rd;
+    uint64_t uid = rd();
+    tmp_path.replace_extension(".slang-module-" + string::hexlify(&uid, sizeof(uid)));
+
```

Cppcheck (2.21.0)

[style] The function 'to_wstring' is never used.

(unusedFunction)